src/fun/builtins.hvm

```diff
@@ -42,3 +45,39 @@ Map/set map key value =
         _: (Map/node value Map/leaf Map/leaf)
       }
   }
+
+// IO Impl
+
+STRING_NIL_TAG  = 0
+STRING_CONS_TAG = 1
+
+IO_DONE_TAG       = 0
+IO_PUT_TEXT_TAG   = 1
+IO_GET_TEXT_TAG   = 2
+IO_WRITE_FILE_TAG = 3
+IO_READ_FILE_TAG  = 4
+IO_GET_TIME_TAG   = 5
+IO_SLEEP_TAG      = 6
+IO_DRAW_IMAGE_TAG = 7
+
+data IO
+  = (Done term)
+  | (PutText   text      cont)
+  | (GetText             cont)
+  | (WriteFile file data cont)
+  | (ReadFile  file      cont)
+  | (GetTime             cont)
+  | (Sleep     time      cont)
+  | (DrawImage tree      cont)
+
+MkStr (String/cons x xs) = λt (t STRING_CONS_TAG x (MkStr xs))
```

Comment:
We also need a function to read strings right? The opposite of this one (matches on the HVM string and converts to scott encoding